lessons/07-mcp/README.md

```diff
@@ -0,0 +1,482 @@
+
+## Narrative: Scipio Africanus
+
+> [!NOTE] 
+> _Our story so far. You are a mechanic from 1860s London. You were working on your automaton and received a letter from Charles Babbage that ended up taking you to a library where you picked up a time travel device. Throughout your travels in time you've ended up in Florence, where you met Leonardo Da Vinci. You also went to the Aztec empire and this is where the story continues._
+>
+> See [Lesson 1](../01-intro-to-genai/README.md) if you want to catch up with the story from the beginning. 
+
+> [!NOTE] 
+> While we recommend going through the story (it's fun!), [click here](#interact-with-scipio) if you'd prefer to jump straight to the technical content.
+
+_Our heroes had just learned tools and using tools via tool calling inspired by their meeting with Amelia Earhart and is now returning back to Ada Lovelace's mansion to discuss their new found knowledge. A flash of light envelops our heroes as the time beetle transports them back to Aaa's present. They find themselves in the basement of Ada Lovelace's mansion, Charles Babbage is not present, but Ada is there waiting for them. She is excited to hear about their journey and the knowledge they've gained._
+
+**Ada**: "Welcome back! I trust your journey was enlightening."
+
+**You**: "It was! meeting with mrs Earhart was quite the experience. She taught us about the importance of tools and how to use them effectively." 
+
+**Ada**: "There are few people who understand the importance of tools better than Amelia. I trust you've also upgraded her time beetle?"
+
+**You**: "Yes, we did! So what's next on the agenda?"
+
+**Ada**: "Tool calling is definitely a powerful concept, one that improves your device. However, it lacks a certain "finesse" shall we say. We need to learn _how to make it scale_, we need a protocol. I know just the fellow to help us with that. His name is Scipio Africanus, a great military strategist from ancient Rome. Too bad he spent all his time with military strategy, I'm sure he would have been a brilliant engineer. Off you go.
+
+**You**: "Scipio Africanus? I think I remember him from history class.  
+
+**Ada**: "Good then, why are you still here? Time is off the essence. Ta ta"
+
+The swirling of light envelops you again and you find yourself in a large room with a large table in the middle. The walls are adorned with maps and military strategies. Scipio Africanus is standing at the table, deep in thought. He looks up as you enter. "Ada te misit/Ada sent you?" 
+
+<div>
+  <img width="600" src="./assets/scipio.png" alt="Scipio Africanus">
+</div>
+
+> Scipio Africanus, also known as Publius Cornelius Scipio Africanus, was a prominent Roman general and statesman who lived from 236 BC to 183 BC. He is best known for his decisive role in the Second Punic War against Carthage and his victory over Hannibal at the Battle of Zama in 202 BC.
+>
+> **Key Highlights of His Life and Career:**
```

Comment:
While I like the Key Highlights, I think the above paragraph provides enough information about him to move on. I'd remove the other information to shorten this up and let people get to the heart of the content sooner.